nixos/modules/services/games/xonotic.nix

```diff
@@ -0,0 +1,197 @@
+{ config
+, pkgs
+, lib
+, ...
+}:
+
+let
+  cfg = config.services.xonotic;
+
+  serverCfg = pkgs.writeText "xonotic-server.cfg" (
+    (if cfg.prependConfig != null then cfg.prependConfig else "")
```

Comment:
`lib.optionalString (cfg.prependConfig != null) cfg.prependConfig`

Reply:
Good point but actually it looks like `toString null` is `""` so I think I can just do `toString cfg.prependConfig`; not sure which is more gooder though

Reply:
I went the `toString` route since it seems more to-the-point than the other two ways. Feel free to unresolve if you'd prefer not to use `toString` for this.

Reply:
Simple is good.